### ISSUE TITLE
I have encountered a fatal mistake and need your help

### ISSUE THREAD

**Issue by suyinchuo** [s3team/DeepFuzz#1]
What parameters should be passed into the "decode_sequence" function?
I ran your code and found that I passed in a string of 50.
But it's not going to work.
So I need your help. Thank you very much.

**Comment by suyinchuo** [comment]
The problem was in the file "generate.py"

**Comment by BambooL** [comment]
> What parameters should be passed into the "decode_sequence" function?
> I ran your code and found that I passed in a string of 50.
> But it's not going to work.
> So I need your help. Thank you very much.

That should be `decode_sequence(input_seq, diversity)`.

**Comment by suyinchuo** [comment]
What kind of "input_seq" is this?Should it be a string of length 50? Thank you very much

**Comment by suyinchuo** [comment]
I think it's probably my problem that's causing the code to fail

**Comment by BambooL** [comment]
> What kind of "input_seq" is this?Should it be a string of length 50? Thank you very much

Yeah, in our experiment, we use string of length 50; the function should handle string of any length, yet results and performance may vary.

**Comment by suyinchuo** [comment]
"States_value = encoder_model.predict(input_seq)", this code keeps reporting errors when I'm running

**Comment by ty5491003** [comment]
> "States_value = encoder_model.predict(input_seq)", this code keeps reporting errors when I'm running

I have the same problem as you. Have you solved it?

**Comment by suyinchuo** [comment]
No, I think there is something wrong with the code

**Comment by lv2020** [comment]
Based on @ty5491003's pr, I change the input_seq from string to numpy array, just like encoder_input_data in train.py. And now it can run. The code is here:
```python
def embed(input_seq):
    encoder_input_data = np.zeros(
    (1, len(input_seq), num_encoder_tokens),
    dtype=np.float16)
    for t, char in enumerate(input_seq.replace('\n',' ')):
        encoder_input_data[0, t, input_token_index[char]] = 1.
    return encoder_input_data
```
And after applying this function, generate.py line 130 should be:
```python
states_value = encoder_model.predict(embed(input_seq))
```
But I didn't get a satisfied result now, I think it's because I only use 100000 data for training. And I don't know whether this change is correct or not. If you have a better solution, please share with me. Thanks!

**Comment by ty5491003** [comment]
@lv2020 Nice work!

**Comment by rain6851** [comment]
> Based on @ty5491003's pr, I change the input_seq from string to numpy array, just like encoder_input_data in train.py. And now it can run. The code is here:
> 
> ```python
> def embed(input_seq):
>     encoder_input_data = np.zeros(
>     (1, len(input_seq), num_encoder_tokens),
>     dtype=np.float16)
>     for t, char in enumerate(input_seq.replace('\n',' ')):
>         encoder_input_data[0, t, input_token_index[char]] = 1.
>     return encoder_input_data
> ```
> 
> And after applying this function, generate.py line 130 should be:
> 
> ```python
> states_value = encoder_model.predict(embed(input_seq))
> ```
> 
> But I didn't get a satisfied result now, I think it's because I only use 100000 data for training. And I don't know whether this change is correct or not. If you have a better solution, please share with me. Thanks!

@lv2020 The speed is to generate test cases.
